### ISSUE TITLE
Fix NSInvalidArgumentException in - [ARGeneViewController loadGene], line 98

### ISSUE THREAD

**Issue by artsyit** [artsy/eigen#599]
### Stacktrace ###

<pre>ARGeneViewController;loadGene;ARGeneViewController.m;98
ARGeneViewController;viewDidLoad;ARGeneViewController.m;65</pre>

### Reason ###

*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[Artwork updateGene:]: unrecognized selector sent to instance 0x129333320'

### Link to HockeyApp ###

* [https://rink.hockeyapp.net/manage/apps/37029/crash_reasons/38546199](https://rink.hockeyapp.net/manage/apps/37029/crash_reasons/38546199)

**Comment by alloy** [comment]
What happens is that `self.gene` [here](https://github.com/artsy/eigen/blob/f92f55841bf30df9b3ee5bad455d548e7879be81/Artsy/Views/Collection_Views/ARGeneViewController.m#L101) is sometimes returning an Artist or Artwork instead of a Gene.

Not sure yet if this is because of some logic elsewhere which is assigning an Artist/Artwork instead of a Gene or if it’s a memory problem.

One memory related logical error I do see with the code is that `self` is being weakified just before calling `-[ARGeneViewController gene]` on it, which means it could theoretically be released from another thread. It should probably be:

```objc
- (void)loadGene
{
    Gene *gene = self.gene;
    @weakify(self);
    [gene updateGene:^{
        // ....
    }];
}
```

**Comment by alloy** [comment]
Based on the breadcrumbs, this appears to be happening when transitioning from the ‘artists’ tab to the ‘categories’ tab in the ‘YOU’ section.

**Comment by alloy** [comment]
This has not been occurring often lately, so moving it out of this milestone.

**Comment by alloy** [comment]
Closed in favour of #599
